corehq/motech/openmrs/forms.py

```diff
@@ -13,3 +22,69 @@ def __init__(self, *args, **kwargs):
+    server_url = forms.CharField(label=_('OpenMRS URL'), required=True,
+                                 help_text=_('e.g. "http://www.example.com/openmrs"'))
+    username = forms.CharField(label=_('Username'), required=True)
+    password = forms.CharField(label=_('Password'), widget=forms.PasswordInput, required=False)
+    log_level = forms.TypedChoiceField(label=_('Log Level'), required=False, choices=LOG_LEVEL_CHOICES, coerce=int)
+    report_uuid = forms.CharField(label=_('Report UUID'), required=True,
+                                  help_text=_('The OpenMRS UUID of the report of patients to be imported'))
+    case_type = forms.CharField(label=_('Case Type'), required=True)
+    owner_id = forms.CharField(label=_('Owner ID'), required=True,
+                               help_text=_('The ID of the mobile worker or location who will own new cases'))
+    import_frequency = forms.ChoiceField(label=_('Import Frequency'), choices=IMPORT_FREQUENCY_CHOICES,
+                                         help_text=_('How often should cases be imported?'), required=False)
+
+    def __init__(self, *args, **kwargs):
+        super(OpenmrsImporterForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper()
+        self.helper.form_class = 'form-horizontal'
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+        self.helper.layout = crispy.Layout(
+            crispy.Fieldset(
+                _('Edit OpenMRS Importer'),
+                crispy.Field('server_url'),
+                crispy.Field('username'),
+                crispy.Field('password'),
+                crispy.Field('log_level'),
+                crispy.Field('report_uuid'),
+                crispy.Field('case_type'),
+                crispy.Field('owner_id'),
+                crispy.Field('import_frequency'),
+            ),
+            hqcrispy.FormActions(
+                StrictButton(
+                    _("Update OpenMRS Importer"),
+                    type="submit",
+                    css_class='btn-primary',
+                )
+            ),
+        )
+
+    def save(self, domain_name):
+        try:
+            importers = get_openmrs_importers_by_domain(domain_name)
+            importer = importers[0] if importers else None  # TODO: Support multiple
+            if importer is None:
+                importer = OpenmrsImporter(domain=domain_name)
+            importer.server_url = self.cleaned_data['server_url']
+            importer.username = self.cleaned_data['username']
+            if self.cleaned_data['password']:
+                # Don't save it if it hasn't been changed. Use simple symmetric encryption. We don't need it to be
+                # strong, considering we'd have to store the algorithm and the key together anyway; it just
+                # shouldn't be plaintext.
+                importer.password = b64encode(bz2.compress(self.cleaned_data['password']))
```

Comment:
It should be possible to use the Django secret key or else the HQ RSA public key to make this more secure. 
@dannyroberts any thoughts on this?

Reply:
(For a little context, I used the same approach to store DHIS2 and OpenClinica passwords, but repeaters store passwords in plaintext (including the OpenMRS repeater). I guess if unfriendly people are reading our database we have more things to worry about than the OpenMRS password, but I do like the idea of using the Django secret key -- it seems a little safer, and still simple.)